src/fn/watch-project.js

```diff
@@ -0,0 +1,177 @@
+                error(err);
+                return;
+            }
+            for (const event of events) {
+                if (!(event.type === 'update' || event.type === 'create')) {
+                    continue;
+                }
+                const changePath = event.path;
+                const parsedPath = path.parse(changePath);
+
+                if (parsedPath.dir.startsWith(appFormsPath)) {
+                    const fileName = parsedPath.base;
+                    if (fileName.match(formXMLRegex) || fileName.match(formPropertiesRegex)) {
+                        eventQueue.enqueue(() => {
+                            return uploadForms(environment.pathToProject, 'app', {
+                                forms: [fileName.split('.')[0]],
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    if (fileName.match(formXLSRegex)) {
+                        eventQueue.enqueue(() => {
+                            return convertForms(environment.pathToProject, 'app', {
+                                enketo: true,
+                                forms: [fileName.split('.')[0]],
+                                transformer: xml => xml.replace('</instance>', '</instance>\n      <instance id="contact-summary"/>'),
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    const formMediaDir = parsedPath.dir.replace(appFormsPath, '');
+                    if (formMediaDir && formMediaDir.match(formMediaRegex)) {
+                        eventQueue.enqueue(() => {
+                            return uploadForms(environment.pathToProject, 'app', {
+                                forms: [formMediaDir.split('-')[0]],
+                            }).then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    continue;
+                }
+
+                if (parsedPath.dir === contactFormsPath) {
+                    const fileName = parsedPath.base;
+                    if (fileName.match(formXLSRegex)) {
+                        eventQueue.enqueue(() => {
+                            return convertContactForm(environment.pathToProject, [fileName.split('.')[0]])
+                                .then(() => Promise.resolve(fileName));
+                        });
+                        continue;
+                    }
+                    if (fileName.match(formXMLRegex)) {
+                        eventQueue.enqueue(() => {
+                            return uploadForms(environment.pathToProject, 'contact', {
+                                id_prefix: 'contact:',
+                                forms: [fileName.split('.')[0]],
+                                default_context: { person: false, place: false },
```

Comment:
As above, can we find a way to reuse the `convertContactForms.execute` here, so we don't duplicate the default_context and id_prefix params?